sw/device/silicon_creator/lib/dice.c

```diff
@@ -319,3 +353,77 @@ rom_error_t dice_cdi_1_cert_build(manuf_certgen_inputs_t *inputs,
 
   return kErrorOk;
 }
+
+rom_error_t dice_tpm_ek_cert_build(manuf_certgen_inputs_t *inputs,
+                                   attestation_public_key_t *tpm_ek_pubkey,
+                                   uint8_t *tpm_ek_tbs, size_t *tpm_ek_tbs_size) {
+  hmac_digest_t digest;
+
+  tpm_ek_tbs_values_t tpm_ek_tbs_params = {
+      .auth_key_key_id = inputs->auth_key_key_id,
+      .auth_key_key_id_size = kDiceCertKeyIdSizeInBytes,
+      .creator_pub_key_ec_x = (uint8_t *)tpm_ek_pubkey->x,
```

Comment:
We cast with `unsigned char *` so as not to violate strict aliasing rules (https://www.open-std.org/jtc1/sc22/wg14/www/docs/n1570.pdf; Section 6.5, Item 7) and for consistency with the above code.

Applies below too.

Reply:
fixed